### PR TITLE
ifacestate/hotplug: updateDevice helper

### PR DESCRIPTION
**Pull request by stolowski**
UpdateDevice helper method for hotplug. The method handles the case where a hotplug device is added again and some of its attributes change, leading to change in slot attributes. The helper creates three tasks:

- hotplug-disconnect runs actual disconnect and disconnect hooks with old attributes for all affected connections
- hotplug-update-slots updates the definiton of the slot in the repo.
- hotplug-connect runs connect and interface hooks to recreate the connection with new attributes.

### CODE REVIEW

File: overlord/ifacestate/hotplug.go [hunk 1]
```diff
@@ -173,3 +174,20 @@ func suggestedSlotName(devinfo *hotplug.HotplugDeviceInfo, fallbackName string)
 	}
 	return shortestName
 }
+
+// create tasks to disconnect slots of given device, update the slot in the repository, then connect it back.
```
Comment (zyga) [inline overlord/ifacestate/hotplug.go]:
```suggestion
// updateDevice creates tasks to disconnect slots of given device, update the slot in the repository and connect it back.
```

File: overlord/ifacestate/hotplug.go [hunk 2]
```diff
@@ -173,3 +174,20 @@ func suggestedSlotName(devinfo *hotplug.HotplugDeviceInfo, fallbackName string)
...
+func updateDevice(st *state.State, ifaceName, hotplugKey string, newAttrs map[string]interface{}) *state.TaskSet {
+	hotplugDisconnect := st.NewTask("hotplug-disconnect", fmt.Sprintf("Disable connections of device %q", hotplugKey))
+	setHotplugAttrs(hotplugDisconnect, ifaceName, hotplugKey)
+
+	updateSlot := st.NewTask("hotplug-update-slot", fmt.Sprintf("Update slot of interface %s, hotplug key %q", ifaceName, hotplugKey))
```
Comment (zyga) [inline overlord/ifacestate/hotplug.go]: Nothing actionable yet but I think we need to see how this looks like in practice once we interact with hot plug and word-smith it into something sensible to the user.

File: overlord/ifacestate/hotplug.go [hunk 3]
```diff
@@ -178,6 +178,23 @@ func suggestedSlotName(devinfo *hotplug.HotplugDeviceInfo, fallbackName string)
 	return shortestName
 }
 
+// updateDevice creates tasks to disconnect slots of given device, update the slot in the repository, then connect it back.
+func updateDevice(st *state.State, ifaceName, hotplugKey string, newAttrs map[string]interface{}) *state.TaskSet {
+	hotplugDisconnect := st.NewTask("hotplug-disconnect", fmt.Sprintf("Disable connections of device %q", hotplugKey))
```
Comment (pedronis) [inline overlord/ifacestate/hotplug.go]: Mmh,  this doesn't mention even the interface,  we should probably try to add something along the concept of a human understandable/showable subset of the hotplugKey. Also probably mention the slot names as well whenever possible.

**Comment by pedronis** [comment]
@zyga this should have not been merged btw, as part of the hotplug main functionality, without me looking at it first

File: overlord/ifacestate/hotplug.go (hunk 3 above)
Comment (stolowski) [reply overlord/ifacestate/hotplug.go]: Small tweak - https://github.com/snapcore/snapd/pull/6315 - to make it consistent with other hotplug summaries. Will revisit once we agree on details.

**Pull request merged**